### PR TITLE
Thomas/feature/81

### LINKED ISSUE #86: RDKIT Error
```Python
    def test_adlercreutzia_equolifaciens_bigg_id_chebi(self):
        # load the e.coli core model and bulk annotate the metabolites. Check if any annoation tkes place (Chebi should cover all metabolites)
        mod_path = self.dat.joinpath("Adlercreutzia_equolifaciens_DSM_19450.xml")
        mod = MeMoModel.fromPath(mod_path)
        exp = AnnotationResult(0,577,196)
        res1 = AnnotationResult.fromTuple(annotateBiGG_id(mod.metabolites))
        res2 = AnnotationResult.fromTuple(annotateChEBI(mod.metabolites))
        add_test_case_to_table(self.test_adlercreutzia_equolifaciens_bigg_id_chebi.__name__, res1 + res2, exp)   
        self.assertLessEqual(exp, res1 + res2, msg=f"Expected amount of annotated metabolites: {exp}, calculated amount of annotated metabolites: {res1 + res2}. All three must be >=")
```

results in

```======================================================================
ERROR: test_adlercreutzia_equolifaciens_bigg_id_chebi (__main__.Test_annotateBulkRoutines.test_adlercreutzia_equolifaciens_bigg_id_chebi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/td/Projects/MeMoMe/tests/test_bulkPerformance.py", line 162, in test_adlercreutzia_equolifaciens_bigg_id_chebi
    res2 = AnnotationResult.fromTuple(annotateChEBI(mod.metabolites))
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/td/Projects/MeMoMe/src/annotateChEBI.py", line 40, in annotateChEBI
    inchi = findOptimalInchi(inchis)
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/td/Projects/MeMoMe/src/annotateInchiRoutines.py", line 66, in findOptimalInchi
    k = k + int(matchMetsByInchi(inchis[i], inchis[j], verbose = verbose)[0])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/td/Projects/MeMoMe/src/matchMets.py", line 46, in matchMetsByInchi
    same = compareInchiByStereoIsomer0(m1,m2, verbose = verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/td/Projects/MeMoMe/src/matchInchiRoutines.py", line 165, in compareInchiByStereoIsomer0
    stereo1 = [Chem.MolToInchi(x) for x in EnumerateStereoisomers(m1, options=opts)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/td/mambaforge/envs/MeMoMe/lib/python3.12/site-packages/rdkit/Chem/EnumerateStereoisomers.py", line 328, in EnumerateStereoisomers
    flippers[i].flip(flag)
  File "/home/td/mambaforge/envs/MeMoMe/lib/python3.12/site-packages/rdkit/Chem/EnumerateStereoisomers.py", line 49, in flip
    self.bond.SetStereo(Chem.BondStereo.STEREOTRANS)
RuntimeError: Pre-condition Violation
	Stereo atoms should be specified before specifying CIS/TRANS bond stereochemistry
	Violation occurred on line 321 in file Code/GraphMol/Bond.h
	Failed Expression: what <= STEREOE || getStereoAtoms().size() == 2
	RDKIT: 2023.09.5
	BOOST: 1_82
```

### PR DESCRIPTION
**Pull request by Unaimend**
Fixes #86 and #81. 

We just iterate over all databases until no change in annotated inchis/dbs/names is detected

### CODE REVIEW

File: src/MeMoModel.py [hunk 1]
```diff
@@ -61,24 +61,33 @@ def fromSBML(cls, model: sbml.Model) -> MeMoModel:
         _id = model.getId()
         return MeMoModel(metabolites=metabolites, _id=_id)
 
-    def annotate(self) -> None:
+    def annotate(self) -> AnnotationResult:
         """Goes through the different bulk annotation methods and tries to annotate InChI strings to the metabolites
         in the model"""
+        print(self._id)
+
```
Comment (Unaimend) [inline src/MeMoModel.py:68]: Currently, we do not use the origin_database here, this should be changed before merge

**Comment by Porthmeus** [comment]
please remove the fix for #86 in this PR, as ignoring the problem is not fixing it.

**Comment by Unaimend** [comment]
> please remove the fix for #86 in this PR, as ignoring the problem is not fixing i

> please remove the fix for #86 in this PR, as ignoring the problem is not fixing it.

This PR is not executable without this "fix"

**Comment by Porthmeus** [comment]
> > please remove the fix for #86 in this PR, as ignoring the problem is not fixing i
> 
> > please remove the fix for #86 in this PR, as ignoring the problem is not fixing it.
> 
> This PR is not executable without this "fix"

Does it spit a warning at least? We should keep the issue open in any case.

**Comment by Unaimend** [comment]
> > > please remove the fix for #86 in this PR, as ignoring the problem is not fixing i
> > 
> > 
> > > please remove the fix for #86 in this PR, as ignoring the problem is not fixing it.
> > 
> > 
> > This PR is not executable without this "fix"
> 
> Does it spit a warning at least? We should keep the issue open in any case.

As discussed, it does

File: src/MeMoModel.py [hunk 2]
```diff
@@ -67,24 +73,45 @@ def fromSBML(cls, model: sbml.Model) -> MeMoModel:
...
+          # BiGG
+          temp_result = annotateBiGG(self.metabolites)
+          #print("BiGG:",temp_result)
+          anno_result = anno_result + temp_result
+          # Use ChEBI
+          temp_result = annotateChEBI(self.metabolites)
+          #print("ChEBI:",temp_result)
+          anno_result = anno_result + temp_result
+          # GO BULK WISE ThORUGH BIGG AND VMH AND MODELSEED, try to extract as much as possible
+          temp_result = annotateModelSEED(self.metabolites)
```
Comment (Porthmeus) [inline src/MeMoModel.py:105]: Looks like here is a `anno_result = anno_result + temp_result` missing after ModelSeed annotation. Please confirm!

**Review by Porthmeus — COMMENTED** [review]
I think I spotted a small bug, but might be an ommission on purpose in line 105-107 for MeMoModel.py - @Unaimend  please confirm.

**Comment by Porthmeus** [comment]
need fixing of  #132 otherwise this will fail

**Comment by Unaimend** [comment]
Test case thats failing 
image
 
 
image
Output of the matching.

You can see that keep1ToMany is false but the output contains one to many matches. I think they have been matched on inchi so the matchInchi function does not properly respect the parameter

**Pull request closed without merging**